centreon/src/Core/TimePeriod/Application/UseCase/AddTimePeriod/AddTimePeriod.php

```diff
@@ -56,6 +69,17 @@ public function __invoke(AddTimePeriodRequest $request, PresenterInterface $pres
         try {
             $this->info('Add a new time period', ['request' => $request]);
 
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_TIME_PERIODS_READ_WRITE)) {
+                $this->error('User doesn\'t have sufficient rights to edit time periods', [
+                    'user_id' => $this->user->getId(),
+                ]);
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(TimeperiodException::editNotAllowed()->getMessage())
+                );
+
+                return;
+            }
+
             if ($this->readTimePeriodRepository->nameAlreadyExists($request->name)) {
```

Comment:
```suggestion
            if ($this->readTimePeriodRepository->nameAlreadyExists(new trimString($request->name))) {
```

Reply:
You can cherry-pick this commit https://github.com/centreon/centreon/commit/8d36fc363dd8bbcb2509ad673f091a7bf6465373 + adapt like in https://github.com/centreon/centreon/commit/05e35b1c21d7fa25b8b858d20a914cd3e54df8f9